### PR TITLE
feat: adding focus state

### PR DESCRIPTION
**Pull request by ramonmulia**
### PR for adding focus state for buttons, text input and nav components

add global CSS classes

### CODE REVIEW

**Comment by BogdanDanilescu** [comment]
I think we still have some focus classes in the following files:

`file-upload.tsx`
`select.tsx`
`text-input.tsx`
`header-menu.tsx`
`file-upload.html`
`select.html`
`text-input.html`
`header-menu.html`

if you do a search with `focus:` in your editor you will see the components above are using generic tailwind classes instead of our custom components

File: packages/react/ds/src/button/button.stories.tsx
Comment (pregno) [inline packages/react/ds/src/button/button.stories.tsx:1]: Could you add a story for each state?

**Review by pregno — COMMENTED** [review]
added a question

File: packages/react/ds/src/button/button.stories.tsx
Comment (ramonmulia) [reply packages/react/ds/src/button/button.stories.tsx:1]: I added a new story for html and react stories that covers all button variants.

**Review by BogdanDanilescu — APPROVED** [review]
LGTM

**Review by pregno — COMMENTED** [review]
The focus state on the link create a layout shift, the text is moving on left by few pixels.
Please check with Lexa; on the link you should probably use box-shadow instead of border.

**Comment by ramonmulia** [comment]
> The focus state on the link create a layout shift, the text is moving on left by few pixels. Please check with Lexa; on the link you should probably use box-shadow instead of border.

Good call, I replaced for box-shadow.
Focus and focus-visible states should be good now.

**Comment by pregno** [comment]
We are getting closer :)
I've noticed that the new focus on inputs (checkboxes, text-input, textarea) doesn't have rounded corners.

**Comment by ramonmulia** [comment]
> We are getting closer :) I've noticed that the new focus on inputs (checkboxes, text-input, textarea) doesn't have rounded corners.

This is intentional, I talked with Lexa she asked to not put them rounded.

**Review by pregno — APPROVED** [review]

**Pull request merged**